client/src/scene/SceneGameArena.ts

```diff
@@ -184,11 +197,47 @@ export class SceneGameArena extends Phaser.Scene {
             moved = this.gameState.moveIfCan(
                 Tetromino.rotateCW // clock wise
             );
+        } else if (this.keys.shift.isDown) {
```

Comment:
similar to the other key press handlers, this needs to check and ignore all spectator input (`this.gameState.playerId != null`)